feat(documentation): adds metadata story (#2511)
swisspost/design-system: packages/documentation/src/stories/foundation/metadata/metadata.docs.mdx

Reviewer: @b1aserlu it would be nice to have code examples right next to the explanations in addition to the example at the end.

import { Meta, Source } from '@storybook/blocks';
import MetadataExample from './metadata-example.html?raw';
import MetadataStories from './metadata.stories';
import { SourceDarkMode } from '../../../../.storybook/preview';

<Meta of={MetadataStories} />

# Metadata

<div className="lead">
  Metadata is used to communicate additional information about your website to webbrowser, accessible devices or other applications.
</div>

The meta-tag has attributes for some general page settings but can also be used to define other metadata using specific names.
On this page you will find links to oter pages that explain some of those names and what they can be used for.

    - [general attributes](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/meta#attributes)
    - [standart names](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/meta/name)

## Viewport

> This meta-tag can be used to manipulate the size and shape of the viewport. [more information](https://developer.mozilla.org/en-US/docs/Web/HTML/Viewport_meta_tag)

Reviewer 2: ```suggestion
This meta-tag can be used to manipulate the size and shape of the viewport. [More information](https://developer.mozilla.org/en-US/docs/Web/HTML/Viewport_meta_tag)
```

## Google and other search engines

> There are some meta-tags that change the way search engines or google in particular treat the website. [more information](https://developers.google.com/search/docs/crawling-indexing/special-tags?hl=en)

Reviewer 2: ```suggestion
There are some meta-tags that change the way search engines or google in particular treat the website. [More information](https://developers.google.com/search/docs/crawling-indexing/special-tags?hl=en)
```

## Open Graph Tags

Open Graph Tags are used to display a more detailed description as well as an image when a link is shared on X (twitter), facebook or in Microsoft Teams.
For X twitter there are also different tags for twitter cards although there is also some support for open graph tags.

- explained by [facebook](https://developers.facebook.com/docs/sharing/webmasters?locale=en_Us)
- explained by [open graph](https://ogp.me/)
- twitter cards and open graph explained by [X used to be Twitter](https://developer.twitter.com/en/docs/twitter-for-websites/cards/guides/getting-started)

## Favicon

> The favicion is the little icon that is displayed on tabs in the webbowsers and it appears in google search.
> [more information](https://dev.to/masakudamatsu/favicon-nightmare-how-to-maintain-sanity-3al7)

Reviewer 2: ```suggestion
The favicon is the little icon that is displayed on tabs in webbowsers. You can see it as the program icon of a website.
[More information](https://dev.to/masakudamatsu/favicon-nightmare-how-to-maintain-sanity-3al7)
```

## Language

> There is a possibility to provide different versions of a webpage in different languages using metadata.

Reviewer 2: ```suggestion
There is the possibility to provide a specific page in different languages using metadata.
```

> Please note that all alternatives and the current page must be listed on each page. [more information](https://developers.google.com/search/docs/specialty/international/localized-versions)

Reviewer 2: ```suggestion
Please note that the current page and its language alternatives must be listed on each of this pages. [More information](https://developers.google.com/search/docs/specialty/international/localized-versions)
```

## Content Security Policy

Instead of using the HTTP header the Content Security Policy can also be definend in a meta-tag in HTML.
In that case it must be the first meta-tag in the header.

> - [general Information](https://developer.mozilla.org/en-US/docs/Web/HTTP/CSP)
> - [more in depth Information](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy)
> - [Impementation Notes and Examples](https://infosec.mozilla.org/guidelines/web_security#content-security-policy)

Reviewer 2: ```suggestion
- [General information](https://developer.mozilla.org/en-US/docs/Web/HTTP/CSP)
- [More in depth information](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy)
- [Impementation notes and examples](https://infosec.mozilla.org/guidelines/web_security#content-security-policy)
```

import { Meta, Source } from '@storybook/blocks';
import MetadataExample from './metadata-example.html?raw';
import MetadataStories from './metadata.stories';
import { SourceDarkMode } from '../../../../.storybook/preview';

<Meta of={MetadataStories} />

# Metadata

<div className="lead">
  Metadata is used to communicate additional information about your website to webbrowser, accessible devices or other applications.
</div>

The meta-tag has attributes for some general page settings but can also be used to define other metadata using specific names.
On this page you will find links to other pages that explain some of those names and what they can be used for.

    - [General attributes](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/meta#attributes)
    - [Standard names](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/meta/name)

> <Source code={MetadataExample.slice(236, 511)} dark={SourceDarkMode} language="html" />

Reviewer: Slicing has to be updated:
<img width="1237" alt="image" src="https://github.com/swisspost/design-system/assets/1659006/4477425a-bdae-424a-90f9-fc99fc241486">

It would be better to have individual snippets that are put together for the final one instead of having to slice the final one. This would allow for changes to the snippets without updating another part of the code.
